backend/app/config.py

```
@@ -75,6 +81,29 @@ def __init__(self) -> None:
+            doc = app_settings.find_one({"key": "llm_config"}, {"_id": 0, "value": 1})
+            if doc and isinstance(doc.get("value"), dict):
+                cfg = doc["value"]
+                # If provider is openrouter but base url is not set, it should use the default
+                return {
+                    "provider": cfg.get("provider", self.LLM_PROVIDER),
+                    "model": cfg.get("model", self.NL_DEFAULT_MODEL),
+                    "api_key": cfg.get("api_key", self.OPENAI_API_KEY)
+                }
+        except Exception:
+            # Fallback to env if DB not available or table doesn't exist yet
+            pass
```

Ruff (0.15.0)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)